Activate Github Actions for the Jupytext repository

Travis-CI is off for maintenance. Maybe it's a good time to give a try to Github actions?

Useful documentation:
- GitHub actions for Python: https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions
- GitHub actions for Miniconda: https://github.com/marketplace/actions/setup-miniconda

I'd like to test both the install with `pip` (`requirements.txt` and `requirements-dev.txt`) and with `conda` (`environment.yml`), at least on a linux worker.

With https://github.com/mwouts/jupytext/commit/27e916c9263695bfc8e1865f3155fab9f67c2d0c I have the following issues:
- Python 3.4 is not available (I will drop it)
- Flake 8 fails on the pip CI:
```

Lint with flake8
4s
##[error]Process completed with exit code 2.
Run # stop the build if there are Python syntax errors or undefined names
  # stop the build if there are Python syntax errors or undefined names
  flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
  all Python files should follow PEP8 (except some notebooks, see setup.cfg)
  flake8 jupytext tests
  # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
  flake8 . --count --exit-zero --max-complexity=10 --statistics
  shell: /bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.6.10/x64
0
/home/runner/work/_temp/98d1db20-f0af-4eba-af95-cb39421c77b0.sh: line 3: syntax error near unexpected token `('
##[error]Process completed with exit code 2.
```
- `conda env create` fails on the conda CI:
```

Run conda env create --file environment.yml
  conda env create --file environment.yml
  shell: /usr/bin/pwsh -command ". '{0}'"

EnvironmentFileNotFound: '/home/runner/work/jupytext/jupytext/environment.yml' file not found

##[error]Process completed with exit code 1.
```

Maybe the last two issues are a sign that I am not running these commands at the root of the `jupytext` repo?

--- a/.github/workflows/continuous-integration-conda.yml
+++ b/.github/workflows/continuous-integration-conda.yml
@@ -29,12 +29,12 @@ jobs:
           ls
       - name: Lint with flake8
         run: |
-        # stop the build if there are Python syntax errors or undefined names
-        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
-        all Python files should follow PEP8 (except some notebooks, see setup.cfg)
-        flake8 jupytext tests
-        # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
-        flake8 . --count --exit-zero --max-complexity=10 --statistics
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+          # all Python files should follow PEP8 (except some notebooks, see setup.cfg)
+          flake8 jupytext tests
+          # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
+          flake8 . --count --exit-zero --max-complexity=10 --statistics
       - name: Install from source (required for the pre-commit tests)
         run: pip install .
       - name: Test with pytest

--- a/.github/workflows/continuous-integration-conda.yml
+++ b/.github/workflows/continuous-integration-conda.yml
@@ -1,0 +1,36 @@
+name: CI (conda)
+on: [push, pull_request]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
+        python-version: [2.7, 3.6, 3.7, 3.8]
+
+    runs-on: ${{ matrix.os }}
+
+    steps:
+      - uses: goanpeca/setup-miniconda@v1
+        with:
+         auto-update-conda: true
+         python-version: ${{ matrix.python-version }}
+      - name: Conda env
+        shell: pwsh
+        run: conda env create --file environment.yml
+      - name: Conda activate
+        run: conda activate jupytext-dev
+      - name: Lint with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+          all Python files should follow PEP8 (except some notebooks, see setup.cfg)
+          flake8 jupytext tests
+          # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
+          flake8 . --count --exit-zero --max-complexity=10 --statistics
+      - name: Test with pytest
+        run: |
+          coverage run --source=. -m py.test
+      - name: Upload coverage
+        run: |
+          coverage report -m

--- a/.github/workflows/continuous-integration-conda.yml
+++ b/.github/workflows/continuous-integration-conda.yml
@@ -1,15 +1,13 @@
 name: CI (conda)
-on: [push, pull_request]
+on: [push]
 
 jobs:
   build:
     strategy:
       matrix:
         os: ['ubuntu-latest', 'macos-latest', 'windows-latest']
-        python-version: [2.7, 3.6, 3.7, 3.8]
-
+        python-version: [2.7, 3.7]
     runs-on: ${{ matrix.os }}
-
     steps:
       - name: Checkout
         uses: actions/checkout@v2

--- a/.github/workflows/continuous-integration-conda.yml
+++ b/.github/workflows/continuous-integration-conda.yml
@@ -17,29 +17,27 @@ jobs:
         uses: goanpeca/setup-miniconda@v1
         with:
           auto-update-conda: true
+          auto-activate-base: false
+          miniconda-version: 'latest'
           python-version: ${{ matrix.python-version }}
+          environment-file: environment.yml
+          activate-environment: jupytext-dev
       - name: Debug
         run: |
           # Show path and content
           pwd
           ls
-      - name: Create conda env
-        run: conda env create --file environment.yml
-      - name: Conda activate
-        run: conda activate jupytext-dev
       - name: Lint with flake8
         run: |
-          # stop the build if there are Python syntax errors or undefined names
-          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
-          all Python files should follow PEP8 (except some notebooks, see setup.cfg)
-          flake8 jupytext tests
-          # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
-          flake8 . --count --exit-zero --max-complexity=10 --statistics
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        all Python files should follow PEP8 (except some notebooks, see setup.cfg)
+        flake8 jupytext tests
+        # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --statistics
       - name: Install from source (required for the pre-commit tests)
         run: pip install .
       - name: Test with pytest
-        run: |
-          coverage run --source=. -m py.test
+        run: coverage run --source=. -m py.test
       - name: Upload coverage
-        run: |
-          coverage report -m
+        run: coverage report -m

--- a/.github/workflows/continuous-integration-conda.yml
+++ b/.github/workflows/continuous-integration-conda.yml
@@ -23,6 +23,7 @@ jobs:
           environment-file: environment.yml
           activate-environment: jupytext-dev
       - name: Lint with flake8
+        shell: pwsh
         run: |
           # stop the build if there are Python syntax errors or undefined names
           flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
@@ -31,8 +32,11 @@ jobs:
           # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
           flake8 . --count --exit-zero --max-complexity=10 --statistics
       - name: Install from source (required for the pre-commit tests)
+        shell: pwsh
         run: pip install .
       - name: Test with pytest
+        shell: pwsh
         run: coverage run --source=. -m py.test
       - name: Upload coverage
+        shell: pwsh
         run: coverage report -m

--- a/.github/workflows/continuous-integration-conda.yml
+++ b/.github/workflows/continuous-integration-conda.yml
@@ -22,11 +22,6 @@ jobs:
           python-version: ${{ matrix.python-version }}
           environment-file: environment.yml
           activate-environment: jupytext-dev
-      - name: Debug
-        run: |
-          # Show path and content
-          pwd
-          ls
       - name: Lint with flake8
         run: |
           # stop the build if there are Python syntax errors or undefined names

--- a/.github/workflows/continuous-integration-conda.yml
+++ b/.github/workflows/continuous-integration-conda.yml
@@ -22,7 +22,7 @@ jobs:
         run: |
           # Show path and content
           pwd
-          tree -L 1
+          ls
       - name: Create conda env
         run: conda env create --file environment.yml
       - name: Conda activate

--- a/.github/workflows/continuous-integration-conda.yml
+++ b/.github/workflows/continuous-integration-conda.yml
@@ -11,12 +11,19 @@ jobs:
     runs-on: ${{ matrix.os }}
 
     steps:
-      - uses: goanpeca/setup-miniconda@v1
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Setup Miniconda
+        uses: goanpeca/setup-miniconda@v1
         with:
-         auto-update-conda: true
-         python-version: ${{ matrix.python-version }}
-      - name: Conda env
-        shell: pwsh
+          auto-update-conda: true
+          python-version: ${{ matrix.python-version }}
+      - name: Debug
+        run: |
+          # Show path and content
+          pwd
+          tree -L 1
+      - name: Create conda env
         run: conda env create --file environment.yml
       - name: Conda activate
         run: conda activate jupytext-dev
@@ -28,6 +35,8 @@ jobs:
           flake8 jupytext tests
           # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
           flake8 . --count --exit-zero --max-complexity=10 --statistics
+      - name: Install from source (required for the pre-commit tests)
+        run: pip install .
       - name: Test with pytest
         run: |
           coverage run --source=. -m py.test

--- a/.github/workflows/continuous-integration-pip.yml
+++ b/.github/workflows/continuous-integration-pip.yml
@@ -1,0 +1,43 @@
+name: CI (pip)
+on: [push, pull_request]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [2.7, 3.4, 3.5, 3.6, 3.7, 3.8]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install -r requirements-dev.txt
+          # install is required for testing the pre-commit mode
+          pip install .[myst] || true
+          # install black if available (Python 3.6 and above), and autopep8 for testing the pipe mode
+          pip install black || true
+          pip install autopep8 || true
+          # install sphinx_gallery and matplotlib if available (may not work on pypy)
+          pip install sphinx_gallery || true
+      - name: Lint with flake8
+        run: |
+          # stop the build if there are Python syntax errors or undefined names
+          flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+          all Python files should follow PEP8 (except some notebooks, see setup.cfg)
+          flake8 jupytext tests
+          # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
+          flake8 . --count --exit-zero --max-complexity=10 --statistics
+      - name: Test with pytest
+        run: |
+          coverage run --source=. -m py.test
+      - name: Upload coverage
+        run: |
+          coverage report -m

--- a/.github/workflows/continuous-integration-pip.yml
+++ b/.github/workflows/continuous-integration-pip.yml
@@ -28,9 +28,6 @@ jobs:
           pip install sphinx_gallery || true
       - name: Lint with flake8
         run: |
-          # Show path and content
-          pwd
-          ls
           # stop the build if there are Python syntax errors or undefined names
           flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
           # all Python files should follow PEP8 (except some notebooks, see setup.cfg)

--- a/.github/workflows/continuous-integration-pip.yml
+++ b/.github/workflows/continuous-integration-pip.yml
@@ -33,7 +33,7 @@ jobs:
           ls
           # stop the build if there are Python syntax errors or undefined names
           flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
-          all Python files should follow PEP8 (except some notebooks, see setup.cfg)
+          # all Python files should follow PEP8 (except some notebooks, see setup.cfg)
           flake8 jupytext tests
           # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
           flake8 . --count --exit-zero --max-complexity=10 --statistics

--- a/.github/workflows/continuous-integration-pip.yml
+++ b/.github/workflows/continuous-integration-pip.yml
@@ -38,6 +38,3 @@ jobs:
       - name: Test with pytest
         run: |
           coverage run --source=. -m py.test
-      - name: Upload coverage
-        run: |
-          coverage report -m

--- a/.github/workflows/continuous-integration-pip.yml
+++ b/.github/workflows/continuous-integration-pip.yml
@@ -1,14 +1,12 @@
 name: CI (pip)
-on: [push, pull_request]
+on: [push]
 
 jobs:
   build:
-
-    runs-on: ubuntu-latest
     strategy:
       matrix:
         python-version: [2.7, 3.5, 3.6, 3.7, 3.8]
-
+    runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v2

--- a/.github/workflows/continuous-integration-pip.yml
+++ b/.github/workflows/continuous-integration-pip.yml
@@ -7,7 +7,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [2.7, 3.4, 3.5, 3.6, 3.7, 3.8]
+        python-version: [2.7, 3.5, 3.6, 3.7, 3.8]
 
     steps:
       - name: Checkout
@@ -30,7 +30,7 @@ jobs:
         run: |
           # Show path and content
           pwd
-          tree -L 1
+          ls
           # stop the build if there are Python syntax errors or undefined names
           flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
           all Python files should follow PEP8 (except some notebooks, see setup.cfg)

--- a/.github/workflows/continuous-integration-pip.yml
+++ b/.github/workflows/continuous-integration-pip.yml
@@ -10,7 +10,8 @@ jobs:
         python-version: [2.7, 3.4, 3.5, 3.6, 3.7, 3.8]
 
     steps:
-      - uses: actions/checkout@v2
+      - name: Checkout
+        uses: actions/checkout@v2
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v1
         with:
@@ -20,8 +21,6 @@ jobs:
           python -m pip install --upgrade pip
           pip install -r requirements.txt
           pip install -r requirements-dev.txt
-          # install is required for testing the pre-commit mode
-          pip install .[myst] || true
           # install black if available (Python 3.6 and above), and autopep8 for testing the pipe mode
           pip install black || true
           pip install autopep8 || true
@@ -29,12 +28,17 @@ jobs:
           pip install sphinx_gallery || true
       - name: Lint with flake8
         run: |
+          # Show path and content
+          pwd
+          tree -L 1
           # stop the build if there are Python syntax errors or undefined names
           flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
           all Python files should follow PEP8 (except some notebooks, see setup.cfg)
           flake8 jupytext tests
           # exit-zero treats all errors as warnings.  The GitHub editor is 127 chars wide
           flake8 . --count --exit-zero --max-complexity=10 --statistics
+      - name: Install from source (required for the pre-commit tests)
+        run: pip install .
       - name: Test with pytest
         run: |
           coverage run --source=. -m py.test

--- a/environment.yml
+++ b/environment.yml
@@ -11,6 +11,8 @@ dependencies:
   - jupyter_contrib_nbextensions
   - pytest
   - pytest-xdist
+  - pytest-cov
+  - codecov
   - pylint
   - flake8
   - black

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -349,6 +349,11 @@ def git_in_tmpdir(tmpdir):
         print(out)
         return out
 
+    git('init')
+    git('status')
+    git('config', 'user.name', 'jupytext-test-cli')
+    git('config', 'user.email', 'jupytext@tests.com')
+
     return git
 
 
@@ -360,8 +365,6 @@ def test_pre_commit_hook(tmpdir):
     nb = new_notebook(cells=[])
 
     git = git_in_tmpdir(tmpdir)
-    git('init')
-    git('status')
     hook = str(tmpdir.join('.git/hooks/pre-commit'))
     with open(hook, 'w') as fp:
         fp.write('#!/bin/sh\n'
@@ -388,8 +391,6 @@ def test_pre_commit_hook(tmpdir):
 def test_sync_with_pre_commit_hook(tmpdir):
     # Init git and create a pre-commit hook
     git = git_in_tmpdir(tmpdir)
-    git('init')
-    git('status')
     hook = str(tmpdir.join('.git/hooks/pre-commit'))
     with open(hook, 'w') as fp:
         fp.write('#!/bin/sh\n'
@@ -472,8 +473,6 @@ def test_pre_commit_hook_in_subfolder(tmpdir):
     nb = new_notebook(cells=[])
 
     git = git_in_tmpdir(tmpdir)
-    git('init')
-    git('status')
     hook = str(tmpdir.join('.git/hooks/pre-commit'))
     with open(hook, 'w') as fp:
         fp.write('#!/bin/sh\n'
@@ -504,8 +503,6 @@ def test_pre_commit_hook_py_to_ipynb_and_md(tmpdir):
     nb = new_notebook(cells=[])
 
     git = git_in_tmpdir(tmpdir)
-    git('init')
-    git('status')
     hook = str(tmpdir.join('.git/hooks/pre-commit'))
     with open(hook, 'w') as fp:
         fp.write('#!/bin/sh\n'
@@ -541,8 +538,6 @@ def test_pre_commit_hook_sync_black_flake8(tmpdir, nb_file):
     metadata = read(nb_file).metadata
 
     git = git_in_tmpdir(tmpdir)
-    git('init')
-    git('status')
     hook = str(tmpdir.join('.git/hooks/pre-commit'))
     with open(hook, 'w') as fp:
         fp.write('#!/bin/sh\n'
@@ -586,8 +581,6 @@ def test_manual_call_of_pre_commit_hook(tmpdir):
         return system(*args, cwd=str(tmpdir))
 
     git = git_in_tmpdir(tmpdir)
-    git('init')
-    git('status')
 
     def hook():
         with mock.patch('jupytext.cli.system', system_in_tmpdir):

--- a/tests/test_using_cli.py
+++ b/tests/test_using_cli.py
@@ -1,11 +1,11 @@
 import os
 import re
-import shutil
 import shlex
 import pytest
 import jupytext
+from nbformat.v4.nbbase import new_notebook, new_code_cell
 from jupytext.cli import jupytext as jupytext_cli
-from .utils import list_notebooks, requires_black
+from .utils import requires_black
 
 doc_path = os.path.join(
     os.path.dirname(os.path.realpath(__file__)),
@@ -14,15 +14,14 @@ doc_path = os.path.join(
 
 @requires_black
 @pytest.mark.skipif(not os.path.isdir(doc_path), reason='Documentation folder is missing')
-@pytest.mark.parametrize('nb_file', list_notebooks('ipynb_py')[-1:])
-def test_jupytext_commands_in_the_documentation_work(tmpdir, nb_file):
+def test_jupytext_commands_in_the_documentation_work(tmpdir):
     # Read the documentation as a bash notebook
     using_cli = os.path.join(doc_path, 'using-cli.md')
     assert os.path.isfile(using_cli)
     using_cli_nb = jupytext.read(using_cli)
 
     # Run the commands in tmpdir on a sample notebook
-    shutil.copy(nb_file, str(tmpdir.join('notebook.ipynb')))
+    jupytext.write(new_notebook(cells=[new_code_cell('1+1')]), str(tmpdir.join('notebook.ipynb')))
     os.chdir(str(tmpdir))
 
     cmd_tested = 0

